transport/x/cherami/config.go

```diff
@@ -0,0 +1,269 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cherami
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"go.uber.org/yarpc/api/transport"
+	"go.uber.org/yarpc/x/config"
+
+	"github.com/uber/cherami-client-go/client/cherami"
+)
+
+const (
+	_destinationSuffix   = "yarpc_dest"
+	_consumerGroupSuffix = "yarpc_cg"
+)
+
+var errNoPeerList = errors.New(
+	"cannot automatically discover Cherami: " +
+		"no peer list provided: " +
+		"please set the peerList attribute or " +
+		"provide a default peer list using the DefaultPeerList option")
+
+// TransportSpecOption configures the Cherami TransportSpec.
+type TransportSpecOption func(*transportSpec)
```

Comment:
That was intentionally left out. We can add it later if necessary, or I can add it to this PR if we know we need to support it already.